srv/salt/_modules/cephdisks.py

```diff
@@ -504,6 +504,81 @@ def assemble_device_list(self):
         return drives
 
 
+def device_(devicename, pathname=None, match=None):
+    """
+    Find all matching symlinks for devicename.
+    """
+    pathname = _pathname_setting(pathname)
+    match = _match_setting(match)
+
+    cmd = (r"find -L {} -samefile {} \( {} \)".format(pathname, devicename, match))
+    _, _stdout, _stderr = __salt__['helper.run'](cmd)
+    if _stdout:
+        _devices = _stdout.split()
+        index = _prefer_underscores(_devices)
+        return _devices[index]
+    return ""
+
+
+def _match_setting(match):
+    """
+    Predence is command line, pillar, then default
+    """
+    if match:
+        log.warning("Using argument {} for match".format(match))
+    else:
+        match = _seek('ceph:modules:cephdisks:device:match'.split(':'), __pillar__)
+        if match:
+            log.warning("Using pillar value {} for match".format(match))
+        else:
+            match = "-name ata* -o -name scsi* -o -name nvme*"
+    return match
+
+
+def _pathname_setting(pathname):
+    """
+    Predence is command line, pillar, then default
+    """
+    if pathname:
+        log.warning("Using argument {} for pathname".format(pathname))
+    else:
+        pathname = _seek('ceph:modules:cephdisks:device:pathname'.split(':'), __pillar__)
+        if pathname:
+            log.warning("Using pillar value {} for pathname".format(pathname))
+        else:
+            pathname = "/dev/disk/by-id"
+    return pathname
+
+
+def _seek(keys, saltdict):
```

Comment:
I think I'd be more comfortable with a non-recursive seek. E.g.
```
def _seek(keys, dict)
  tmp = dict
  for key in keys:
    if key in tmp:
      tmp = tmp[key]
    else:
      return None
  return tmp
```
Personally I find that easier to read...though I suppose thats nit picking.